src/datahike/spec.cljc

```diff
@@ -0,0 +1,95 @@
+(ns datahike.spec
+  (:require [datahike.datom :refer [datom?]]
+            [datahike.db.utils :as dbu]
+            [spec-tools.data-spec :as ds]
+            [clojure.spec.alpha :as s])
+  #?(:clj
+     (:import [clojure.lang IAtom]
+              [java.util Date])))
+
+(defn set-of [pred] (s/every pred :kind set?))
+
+(defn date? [d]
+  #?(:cljs (instance? js/Date d)
+     :clj  (instance? Date d)))
+
+(def time-point? (s/or :int int? :date date?))
+
+;; TODO: there's lots of runtime logic to conform these transactions - it would
+;; be nice to spec them accurately, and then conform them at runtime through spec
+(def STransactions
+  (s/coll-of (s/or :seq coll? :map map? :nil nil?)))
+
+;; TODO: deduplicate this Spec with the one in datahike.config
```

Comment:
Didn't feel worth it making an issue. Simply using the spec from config namespace now.

Reply:
Alright, you didn't want to move the spec over to spec-ns?

Reply:
I think that topic requires a bit more discussion. There are specs sprayed all over the place, not only the config namespace, which has quite a goo amount of them. The old specs also look different then the ones in the spec namespace. So there is a conversation to be had about which scheme we want to follow to express specs and where they should live. In some instances it might make sense to have them in separate namespaces, e.g. for the index structures, but in others we might want to move them to the spec ns.